lib/src/protos/git_store.proto

```diff
@@ -16,10 +16,22 @@ syntax = "proto3";
 
 package git_store;
 
+message TreeConflict {
+  repeated bytes removes = 1;
+  repeated bytes adds = 2;
+}
+
 message Commit {
   repeated bytes predecessors = 2;
   bytes change_id = 4;
 
+  oneof root_tree {
+    bytes resolved = 3;
+    TreeConflict conflict = 10;
+  }
+  // TODO(#1624): delete when we assume that all commits use this format
+  bool tree_conflict_format = 1;
+
```

Comment:
I'm not sure if these numbers can be reused. If they were stored in file before da5db27bb010, maybe they shouldn't? Ignore this comment if they aren't.

Reply:
They were never stored before then. We only ever stored `predecessors`,`change_id`, `is_open`, and `is_pruned` in the git backend (which had tag numbers 2, 4, 8, 9).

Speaking of this, since we already plan to remove `tree_conflict_format`, I've given number 1 to `conflict` instead. It doesn't really matter, just looks a little nicer :)